packages/netlify/test/functions/edge-middleware.test.js

```diff
@@ -39,7 +39,7 @@ describe('Middleware', () => {
 			expect(contents.includes('"Hello world"')).to.be.false;
 		});
 
-		it('does not apply middleware during prerendering', async () => {
+		it.skip('does not apply middleware during prerendering', async () => {
```

Comment:
Ah yeah, this behavior is maybe confusing, but the current intention is for middleware to run on prerendered _and_ dynamically rendered requests.

We probably need a better way to control this from `astro` itself rather than having it be an adapter decision.

Reply:
Totally agree that this should be controlled from Astro itself - it's an essential piece of how rendering and middleware interact, and that shouldn't live in an adapter that's different for every provider.

When we did the rewrite of this plugin, the Astro docs weren't yet clear about the intended behaviour on this. @lilnasy you mention there was a bug fix, could you link me to more info on this?

My 2cts is that i'd like to remove the edgeMiddleware field from the Adapter options, and have Astro dictate the behaviour here. @natemoo-re you're referring to "the current intention" - is the behaviour around this in-flux? If yes, where can I read up on that so we make sure that the adapter behaves correctly?

The docs on this say "Middleware allows you to intercept requests and responses and inject behaviors dynamically every time a page or endpoint is about to be rendered.", which I find to be a bit fuzzy. Personally, I'd prefer the behaviour to be defined a little more explicit here. Is it "dynamically", or is it "on render"? "dynamically" would mean "on every request" to me, "on render" would mean "depends on your prerender option".

Reply:
> My 2cts is that i'd like to remove the edgeMiddleware field from the Adapter options, and have Astro dictate the behaviour here.

That was the original implementation. We switched letting the adapter decide it in Astro 3.

I don't think we ever sat down to discuss pre-rendered pages with edgeMiddleware enabled, but even from the test case, it comes across as a bug that pages wouldn't have access to locals.

Also I should clarify, when I said this behavior isn't the adapter's to control, I meant that is currently the case - the netlify adapter did nothing to hide locals before, and nothing to add them since Astro 4.2.